solr/core/src/test/org/apache/solr/handler/admin/CoreAdminHandlerTest.java

```diff
@@ -377,7 +378,8 @@ public void testUnloadForever() throws Exception {
     runner.start();
 
     try (SolrClient client =
-        new HttpSolrClient.Builder(runner.getBaseUrl() + "/corex")
+        new HttpSolrClient.Builder(runner.getBaseUrl().toString())
```

Comment:
i wonder if the Builder should just take a Url?    Does we end up doing a .toString a million times?

Reply:
i see if everywhere now that we are losing the type coercion by appending the collection or core name!

Reply:
Yeah, I wondered that as well, lots of URL->String logic.

I had similar thoughts about `Replica`...tons of places where we create a HSC using the baseUrl and coreName pulled out of a `Replica` object.

But both of those are probably beyond the scope of this PR.

Reply:
makes sense...